service/history/api/recordworkflowtaskstarted/api.go

```diff
@@ -167,10 +167,24 @@ func Invoke(
 				return nil, serviceerrors.NewObsoleteMatchingTask("wrong task queue type")
 			}
 
+			wfBehavior := mutableState.GetEffectiveVersioningBehavior()
```

Comment:
is there any opportunity to share code between this and recordactivitytaskstarted? actually I'm not even sure that would make this easier to understand, maybe it's simpler to just do it inline in both. (especially after future cleanups.) just wondering.